synapse/handlers/federation.py

```diff
@@ -889,6 +891,75 @@ async def _process_received_pdu(
                     "resync_device_due_to_pdu", self._resync_device, event.sender
                 )
 
+        await self._handle_marker_event(origin, event)
+
+    async def _handle_marker_event(self, origin: str, marker_event: EventBase):
+        """Handles backfilling the insertion event when we receive a marker
+        event that points to one.
+
+        Args:
+            origin: Origin of the event. Will be called to get the insertion event
+            marker_event: The event to process
+        """
+
+        if marker_event.type != EventTypes.MSC2716_MARKER:
```

Comment:
We should also check the event hasn't been rejected via `marker_event.rejected_reason is None`

Reply:
Added 👍